packages/react-query/src/shared/hooks/createHooksInternal.tsx

```diff
@@ -319,33 +331,76 @@ export function createRootHooks<
         return;
       }
       let isStopped = false;
+
       const subscription = client.subscription(
         path.join('.'),
         input ?? undefined,
         {
+          onConnecting: () => {
+            if (!isStopped) {
+              optsRef.current.onConnecting?.();
+              setSubscriptionState((prev) => ({
+                ...prev,
+                state: 'connecting',
+                isConnecting: true,
+                isConnected: false,
+              }));
+            }
+          },
           onStarted: () => {
             if (!isStopped) {
               optsRef.current.onStarted?.();
+              setSubscriptionState((prev) => ({
+                ...prev,
+                state: 'connected',
+                isConnecting: false,
+                isConnected: true,
+              }));
             }
           },
           onData: (data) => {
             if (!isStopped) {
+              setSubscriptionState((prev) => ({
```

Comment:
I'm not sure we wanna update data for every event - it can become incredibly expensive for high throughput events, likely better to force developers to handle data themselves

Reply:
If we update data of every event, we need to have a smart handler that won't update data unless it's used - not looked at the magic that RQ does for that

Reply:
I probably agree, I'll reconsider, removing it for now

Reply:
Re-added this :)